Issue #669 ability to run tests on postgre sql database

GrKolychev opened the pull request:
# Description

## Background

This PR brings ability to run tests on postgreSQL database, related to issue #669 [](https://github.com/epam/NGB/issues/669)

## Changes

- Created /docker/postgres/docker-compose.yml file to deploy the postgres image via docker
- Changed server/catgenome/build.gradle:
    - added avast [plugin](https://github.com/avast/gradle-docker-compose-plugin) dependency
    - added script, which automatically compose up postgre image before run tests and compose down after.
- Added server/catgenome/profiles/postgres/test-catgenome-acl.properties, server/catgenome/profiles/postgres/test-catgenome-auth.properties, server/catgenome/profiles/postgres/test-catgenome-cache-disable.properties files
## Acceptance criteria

Run tests with -Pdatabase=postgres argument

# Check list

- [ ] Unit tests are provided
- [ ] Integration tests are provided (if CLI/API was changed)
- [ ] Documentation is updated

GrKolychev committed:
#669 Created test-catgenome-acl.properties, test-catgenome-auth.properties, test-catgenome-cache-disable.properties in server/catgenome/profiles/postgres

GrKolychev committed:
#669 Created test-catgenome-acl.properties, test-catgenome-auth.properties, test-catgenome-cache-disable.properties in server/catgenome/profiles/postgres

GrKolychev committed:
created docker-compose.yml

GrKolychev committed:
deleted Dockerfile

GrKolychev committed:
Merge remote-tracking branch 'origin/#669_Ability_to_run_tests_on_postgreSQL_database' into #669_Ability_to_run_tests_on_postgreSQL_database

GrKolychev committed:
#669 edited build.gradle(:server:catgenome) to allow run tests on postgreSQL

GrKolychev committed:
#669 deleted empty line from docker-compose.yml

GrKolychev committed:
Delete .idea directory

GrKolychev committed:
Delete server/catgenome/${homologene.index.directory} directory

GrKolychev committed:
Delete server/catgenome/${taxonomy.index.directory} directory

GrKolychev committed:
Update test-catgenome-cache-disable.properties

SilinPavel commented on docker/postgres/docker-compose.yml:
```diff
@@ -0,0 +1,13 @@
+version: "3.9"
```
please add a license here and for other new file (you can find example of the license in the other files of the project)

SilinPavel commented on docker/postgres/docker-compose.yml:
```diff
@@ -0,0 +1,13 @@
+version: "3.9"
+
```
Lets also add a README.md file here that will describe for what reason we have this files here and how it works

SilinPavel replied:
I would prefer to rename folder `postgres` to something like `postgres_test` and also add some words to docker/README.md what this docker for

GrKolychev committed:
#669
- Created docker/postgres_test/README.md
- Mentioned postgres_test in docker/README.md
- Added licenses to all new files

GrKolychev committed:
#669
Fixed path to docker-compose file in server/catgenome/build.gradle
